### PR TITLE
fix(web): keep unrealized gain ranking limit on account selection

### PR DESCRIPTION
**Pull request by hiroppy**
# Summary

- 含み損益ランキングの表示上限を損益フィルターだけで決定するよう修正
- 個別金融機関を選択しても「損益すべて」は各ランキング3件を維持する回帰テストを追加
- 「含み益」「含み損」は従来どおり対象ランキングを最大6件表示

## Linear Issue

- [HIR-179](https://linear.app/hiroppy/issue/HIR-179/口座選択時に含み損益損益すべてでもランキングが最大6件表示される)

### Selected Quality Characteristics

| Quality characteristic | Why it is relevant | Acceptance criteria |
| ---------------------- | ------------------ | ------------------- |
| 機能適合性 | 口座と損益フィルターの組合せで表示上限が変わる不具合のため | 「損益すべて」は口座に依存せず各3件、「含み益」「含み損」は最大6件 |
| 保守性 | 表示上限の条件が無関係な口座フィルターへ依存していたため | 上限決定が損益フィルターだけから読み取れる |

### Test Techniques

| Test technique | Selection rationale | Expected coverage |
| -------------- | ------------------- | ----------------- |
| 決定表テスト | 損益区分と金融機関選択の組合せを確認するため | `all/gain/loss` と個別金融機関選択時の3件・6件切替 |
| 境界値分析 | 表示上限直前・直後の銘柄を確認するため | all で4件目を非表示、gain で6件目を表示し7件目を非表示 |
| 探索的テスト | 実際の Select 操作と描画を確認するため | デモDBの貸借対照表で個別金融機関選択後の表示を確認 |

### Primary Test Conditions

- 「損益すべて」かつ個別金融機関で各ランキングが3件以内
- 「含み益」かつ個別金融機関で6件目まで表示し7件目は非表示
- `all/gain/loss` の上限がそれぞれ `3/6/6`

### Out-of-Scope Risks

- N/A — データ取得・集計・スタイルは変更していない

## Verification

- [x] Unit tests
- [x] Type checking
- [x] Lint
- [ ] Storybook tests
- [ ] E2E tests
- [x] Manual verification

### Commands Executed

```text
pnpm --filter @mf-dashboard/web exec vitest run src/components/info/holdings-table.client.test.tsx — 24 passed
pnpm --filter @mf-dashboard/web test:unit — 711 passed
pnpm turbo typecheck — 8/8 packages passed
pnpm lint — passed
pnpm format:check — passed
pnpm --filter @mf-dashboard/db build:demo — passed
pnpm --filter @mf-dashboard/web dev --hostname 127.0.0.1 + Playwright walkthrough — passed
```

### Checks Not Run

- Storybook tests — Story とスタイルを変更しておらず、コンポーネント unit test と実アプリ確認で対象条件を網羅
- E2E tests — 対象条件は匿名データのコンポーネントテストで決定的に検証し、同じ操作をデモアプリで手動確認

## Runtime Verification

「損益すべて」で個別金融機関を選択し、含み益上位が3件に維持されることを確認。

HIR-179 runtime verification

## Summary by CodeRabbit

* **Bug Fixes**
  * Standardized holdings ranking limits across institution and category filters.
  * All holdings now display up to 3 ranked items, while gain and loss views display up to 6.
  * Selecting an institution no longer changes the ranking limit for gain-based views.

* **Tests**
  * Added coverage for ranking limits across holdings, gain, loss, and institution-filtered views.

### CODE REVIEW

File: apps/web/src/components/info/holdings-table.client.test.tsx [hunk 1]
```diff
@@ -323,12 +323,43 @@ describe("UnrealizedGainCardClient", () => {
...
+        />
+      </HoldingsFilterProvider>,
+    );
+
+    fireEvent.click(screen.getByRole("button", { name: "金融機関 A に絞る" }));
+    expect(screen.queryByText("含み益銘柄 4")).toBeNull();
+
+    fireEvent.click(screen.getByRole("button", { name: "含み益に絞る" }));
+    expect(screen.getByText("含み益銘柄 6")).not.toBeNull();
+    expect(screen.queryByText("含み益銘柄 7")).toBeNull();
```
[the remark this answers is not in the document]
Comment (hiroppy) [reply apps/web/src/components/info/holdings-table.client.test.tsx:362]: The PR description already includes the requested QA coverage: ISO/IEC 25010 functional suitability and maintainability, ISTQB decision-table/boundary-value/exploratory techniques, primary conditions, out-of-scope risks, and the exact validation commands. CodeRabbit’s own Description check also confirms these sections are present, so no code or description change is needed.

**Comment by hiroppy** [comment]
The runtime screenshot was generated exclusively from the anonymous demo database: `pnpm --filter @mf-dashboard/db build:demo` created `data/demo.db`, then the app ran in `DEMO_MODE=true`. `data/moneyforward.db` and real Money Forward data were not used. The displayed amounts and holdings are synthetic demo seed values, so the artifact already satisfies the requested demo-only condition.

**Review by hiroppy — COMMENTED** [review]
画像はコミットしなくていい

**Comment by hiroppy** [comment]
Commit: https://github.com/hiroppy/mf-dashboard/commit/a5d9e5009029936728ade69d389987d32a694400

## QA

QA story: Functional suitability + interaction capability. Techniques: decision-table testing, boundary value analysis, exploratory testing.

### SBI証券選択時も「損益すべて」は含み益上位3件を維持する

期待値: 金融機関をSBI証券に絞っても、損益フィルターが「損益すべて」の場合は含み益上位が3件までに制限される。

pr363-sbi-all-ranking

### SBI証券選択後に「含み益」へ切り替えるとランキング上限が最大6件に広がる

期待値: 同じSBI証券フィルターのまま「含み益」を選ぶと、該当する含み益銘柄が3件で打ち切られず、最大6件まで表示される。

pr363-sbi-gain-ranking
